src/main/java/com/kmbl/InventoryManagementService/services/InventoryService.java

```diff
@@ -77,4 +77,69 @@ public void freeInventory(CancelOrderMessage message) throws ServiceException {
     public void deleteInventory(String id) {
         inventoryRepository.deleteById(id);
     }
+
+
+
+
+    public ResponseBody updateInventoryforOrder(List<OrderRequestBody> orderRequestItems) {
+        //get list of inventory
+        synchronized (this) {
+            InventoryOrderStatus overallStatus = InventoryOrderStatus.FAILED_ORDER;
+            //  List<ResponseObject> tempResponse=new ArrayList<>();
+            ResponseBody response = new ResponseBody();
+            for (int i = 0; i < orderRequestItems.size(); i++) {
+
+                ResponseObject responseObject = new ResponseObject();
+                List<Inventory> tempInventory = inventoryRepository.findByproductId(orderRequestItems.get(i).getProductID());
```

Comment:
OrderItem contains productId and Seller Id user has already, choose from which seller to buy, dont fetch the data for product Id , just fetch it by sellerId and productId and update inventory accordingly